### PR TITLE
google_rtc_audio_processing: Make control write reliable

### PR DESCRIPTION
**Pull request by bzhg**
Userspace may write the config control repeatedly in quick succession.
Accept the new config in cmd callback to avoid -EBUSY error and ensure
the last successful control write before prepare/copy is applied.

This patch also fixes an edge case during init where the first non-empty
control write happens after prepare. This patch ensures the config is
applied during copy.

Tested:
1) Writes in quick succession
for i in $(seq 1 100); do
    ./sof-ctl -n 92 -t 0 -b -r -s /root/test_blob0.txt &
    ./sof-ctl -n 92 -t 0 -b -r -s /root/test_blob1.txt &
done

2) Large config requiring multiple ipcs
for i in $(seq 1 1000); do
    ./sof-ctl -n 92 -t 0 -b -r -s /root/large_blob.txt
    ./sof-ctl -n 92 -t 0 -b -r -s /root/test_blob1.txt
done

3) Reboot, write before and after arecord starts
4) Suspend/resume

Signed-off-by: Ben Zhang <benzh@chromium.org>

### CODE REVIEW

File: src/audio/google_rtc_audio_processing.c [hunk 1]
```diff
@@ -122,10 +124,23 @@ static int google_rtc_audio_processing_cmd_set_data(
...
+		ret = comp_data_blob_set_cmd(cd->tuning_handler, cdata);
+		if (ret)
+			return ret;
+		/* Accept the new blob immediately so that userspace can write
+		 * the control in quick succession without error.
+		 * This ensures the last successful control write from userspace
+		 * before prepare/copy is applied.
+		 */
+		if (comp_is_new_data_blob_available(cd->tuning_handler)) {
+			comp_get_data_blob(cd->tuning_handler, NULL, NULL);
```
Comment (cujomalainey) [inline src/audio/google_rtc_audio_processing.c:140]: Are we certain that we are not holding an references to the previous data at this point? The old data is freed here so if we did not dup that will be dangerous

Comment (bzhg) [reply src/audio/google_rtc_audio_processing.c:144]: Added comment to explain why it is safe to free the old blob.

**Review by lkoenig — APPROVED** [review]
This look good to me.
I am still a bit itchy about the preemption but as it is possible at the moment, it should be fine.

**Comment by lkoenig** [comment]
I wonder if it would be possible in a follow-up CL to add locks around the config for the preemption case ?

**Review by lgirdwood — APPROVED** [review]

**Review by cujomalainey — APPROVED** [review]

**Pull request merged**